src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -156,13 +156,19 @@ private IEnumerable<ConstructorSignature> BuildPrimaryConstructors(IReadOnlyList
 
             if (Fields.CredentialFields.Count == 0)
             {
-                yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalToRequired).ToArray());
+                /* order the constructor paramters as (endpoint, requiredParameters, OptionalParamters).
+                 * use the OrderBy to put endpoint as the first parameter.
+                 * */
+                yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalToRequired).OrderBy(parameter => !parameter.Name.Equals("endpoint", StringComparison.OrdinalIgnoreCase)).ToArray());
```

Comment:
We have a lot of places directly use "endpoint". I would suggest make it a constant variable.

Reply:
actually we could reference the `KnownParameters` where we have a parameter created for endpoint

Reply:
Yes. Maybe we need to file a MQ task to update `endpoint` to `KnowParameters.Endpoint.Name`. And after migrate mgp to InputModel, we will update this together.